dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/RumFeature.kt

```diff
@@ -117,26 +122,27 @@ internal object RumFeature : SdkFeature<RumEvent, Configuration.Feature.RUM>() {
 
     private fun initializeVitalMonitors() {
         vitalExecutorService = ScheduledThreadPoolExecutor(1)
-        val cpuReaderRunnable = VitalReaderRunnable(
-            CPUVitalReader(),
-            cpuVitalMonitor,
-            vitalExecutorService,
-            VITAL_UPDATE_PERIOD_MS
-        )
-        vitalExecutorService.schedule(
-            cpuReaderRunnable,
-            VITAL_UPDATE_PERIOD_MS,
-            TimeUnit.MILLISECONDS
-        )
 
-        val memoryReaderRunnable = VitalReaderRunnable(
-            MemoryVitalReader(),
-            memoryVitalMonitor,
+        initializeVitalMonitor(CPUVitalReader(), cpuVitalMonitor)
+        initializeVitalMonitor(MemoryVitalReader(), memoryVitalMonitor)
+
+        val frameRateReader = FrameRateVitalReader()
+        initializeVitalMonitor(frameRateReader, frameRateVitalMonitor)
+        Choreographer.getInstance().postFrameCallback(frameRateReader)
```

Comment:
Maybe we should create separated singleton dedicated objects for these monitors and have this methods inside and here we can just call: `CpuVitalMonitor.init()`, `FrameVitalMonitor.init()`, etc. This way you can encapsulate all this logic especially the one specific to a monitor like: `Choreographer.getInstance().postFrameCallback(frameRateReader)`.

Reply:
I'm really not a fan of adding more singleton than we already have especailly since down the line the goal will be to remove singleton objects as much as we can